bigquery/storage/managedwriter/retry.go

```diff
@@ -49,15 +58,62 @@ func (r *defaultRetryer) Retry(err error) (pause time.Duration, shouldRetry bool
 	}
 }
 
+func (r *defaultRetryer) RetryAppend(err error, attemptCount int) (pause time.Duration, shouldRetry bool) {
+	if err == nil {
+		return 0, false // This shouldn't need to be here, and is only provided defensively.
+	}
+	if attemptCount > defaultAppendRetries {
+		return 0, false // exceeded maximum retries.
+	}
+	// This predicate evaluates the received response to determine if we should re-enqueue.
+	apiErr, ok := apierror.FromError(err)
+	if !ok {
+		// These are non status-based errors.
+		// Context errors are non-retriable.
+		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+			return 0, false
+		}
+		// The same errors that trigger fast reconnect are retryable, as they deal with transient problems
+		// with the network stream.
+		if shouldReconnect(err) {
+			return r.bo.Pause(), true
+		}
+		// Any other non-status based errors are not retried.
+		return 0, false
+	}
+	// Next, evaluate service-specific error details.
+	se := &storagepb.StorageError{}
+	if e := apiErr.Details().ExtractProtoMessage(se); e == nil {
+		if se.GetCode() == storagepb.StorageError_OFFSET_OUT_OF_RANGE {
```

Comment:
Why out of range is returning true?

Reply:
See the accompanying doc, shared with you internally.

Reply:
Per discussion, removed this.